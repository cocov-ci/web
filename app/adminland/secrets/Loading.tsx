'use client'

import Loading from 'app/common/Loading'

import styles from './Item/Item.module.scss'

const LoadingItem = () => {
  return (
    <div className={styles.base}>
      <div className={styles.iconWrapper}>
        <Loading alignment="left" height="17px" type="skeleton" width="17px" />
      </div>
      <div className={styles.infoWrapper}>
        <div>
          <Loading
            alignment="left"
            height="15px"
            type="skeleton"
            width="100px"
          />
        </div>
        <div>
          <Loading
            alignment="left"
            height="15px"
            type="skeleton"
            width="300px"
          />
        </div>
        <div>
          <Loading
            alignment="left"
            height="15px"
            type="skeleton"
            width="150px"
          />
        </div>
      </div>
      <div className={styles.ButtonWrapper}>
        <Loading alignment="left" height="35px" type="skeleton" width="35px" />
      </div>
    </div>
  )
}

const LoadingItems = () => (
  <div>
    {new Array(6).fill(0).map(item => (
      <LoadingItem key={item} />
    ))}
  </div>
)

export default LoadingItems
