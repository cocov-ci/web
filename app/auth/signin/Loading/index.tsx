import { RefreshCw } from 'lucide-react'

import styles from './Loading.module.scss'

const Loading = () => {
  return (
    <div className={styles.loading}>
      <RefreshCw className={styles.spinner} />
    </div>
  )
}

export default Loading
