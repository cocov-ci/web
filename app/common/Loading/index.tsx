'use client'

import classNames from 'classnames'
import { LucideIcon, RefreshCw } from 'lucide-react'
import Skeleton, { SkeletonTheme } from 'react-loading-skeleton'

import 'react-loading-skeleton/dist/skeleton.css'
import styles from './Loading.module.scss'
import Spinner from './Spinner'

interface LoadingProps extends SpinnerProps, SkeletonProps {
  className?: string
  alignment?: 'center' | 'left' | 'right'
  type?: 'spinner' | 'skeleton'
  tiny?: boolean
}

interface SpinnerProps {
  size?: number
  spinnerIcon?: LucideIcon | typeof Spinner
}

interface SkeletonProps {
  count?: number
  width?: string
  height?: string
}

const SkeletonComponent = ({ count, height, width }: SkeletonProps) => {
  return (
    <SkeletonTheme baseColor="#e4e4e4" highlightColor="#e9e9e9">
      <Skeleton count={count} height={height} width={width} />
    </SkeletonTheme>
  )
}

const SpinnerComponent = ({ size, spinnerIcon = Spinner }: SpinnerProps) => {
  const Component = spinnerIcon

  return <Component className={styles.spinner} size={size} />
}

const Loading = ({
  width = '300px',
  height = '15px',
  count = 1,
  type = 'spinner',
  alignment = 'center',
  size = 24,
  spinnerIcon,
  className,
  tiny,
}: LoadingProps) => {
  return (
    <div
      className={classNames(styles.loading, styles[alignment], className, {
        [styles.tiny]: tiny,
      })}
    >
      {type === 'spinner' && (
        <SpinnerComponent size={size} spinnerIcon={spinnerIcon} />
      )}
      {type === 'skeleton' && (
        <SkeletonComponent count={count} height={height} width={width} />
      )}
    </div>
  )
}

export default Loading
