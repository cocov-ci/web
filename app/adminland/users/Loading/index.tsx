import Loading from 'app/common/Loading'

import styles from '../Item/Item.module.scss'

export const LoadingItem = () => {
  return (
    <div className={styles.base}>
      <div className={styles.avatarWrapper}>
        <Loading alignment="left" height="25px" type="skeleton" width="25px" />
      </div>
      <div className={styles.dataWrapper}>
        <div className={styles.loginContainer}>
          <Loading
            alignment="left"
            height="15px"
            type="skeleton"
            width="50px"
          />
        </div>
        <div className={styles.description}>
          <Loading
            alignment="left"
            height="15px"
            type="skeleton"
            width="100px"
          />
          <Loading
            alignment="left"
            height="15px"
            type="skeleton"
            width="127px"
          />
          <Loading
            alignment="left"
            height="15px"
            type="skeleton"
            width="155px"
          />
        </div>
      </div>
      <div className={styles.ButtonWrapper}>
        <Loading alignment="left" height="35px" type="skeleton" width="35px" />
      </div>
    </div>
  )
}
