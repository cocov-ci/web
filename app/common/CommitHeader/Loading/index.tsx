import classNames from 'classnames'
import React from 'react'

import Loading from 'app/common/Loading'
import randomBetween from 'utils/randomBetween'

import styles from '../CommitHeader.module.scss'

export type CommitHeaderProps = {
  className?: string
}

const LoadingComponent = ({ className }: CommitHeaderProps) => {
  return (
    <div className={classNames(styles.base, className)}>
      <div className={styles.avatarWrapper}>
        <Loading
          alignment="left"
          className={styles.skeleton}
          height="25px"
          type="skeleton"
          width="25px"
        />
      </div>

      <div className={styles.infoWrapper}>
        <div className={styles.commitMeta}>
          <Loading
            alignment="left"
            height="18px"
            type="skeleton"
            width={`${randomBetween(206, 400)}px`}
          />
        </div>
        <div className={styles.commitTitle}>
          <Loading
            alignment="left"
            height="18px"
            type="skeleton"
            width={`${randomBetween(32, 400)}px`}
          />
        </div>
      </div>
    </div>
  )
}

export default LoadingComponent
