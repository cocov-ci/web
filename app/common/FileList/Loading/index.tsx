'use client'

import classNames from 'classnames'
import React from 'react'

import Loading from 'app/common/Loading'

import styles from '../FileList.module.scss'

export type CommitHeaderProps = {
  className?: string
}

const LoadingComponent = ({ className }: CommitHeaderProps) => {
  return (
    <div className={classNames(styles.base, className)}>
      <div className={styles.header}>
        <div className={styles.file}>
          <Loading
            alignment="left"
            height="24px"
            type="skeleton"
            width="100px"
          />
        </div>
        <div className={styles.coverage}>
          <Loading
            alignment="left"
            height="24px"
            type="skeleton"
            width="100px"
          />
        </div>
      </div>

      <div className={styles.fileList}>
        {Array.from(Array(11).keys()).map(item => (
          <div key={item}>
            <div className={styles.fileItem}>
              <Loading height="28px" type="skeleton" width="100%" />
            </div>
            <div className={styles.fileItem} />
          </div>
        ))}
      </div>
    </div>
  )
}

export default LoadingComponent
