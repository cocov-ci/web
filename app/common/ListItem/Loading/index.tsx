import classNames from 'classnames'

import Box from 'app/common/Box'
import styles from 'app/common/ListItem/ListItem.module.scss'
import Loading from 'app/common/Loading'
import randomBetween from 'utils/randomBetween'

const LoadingListItem = () => {
  return (
    <Box className={classNames(styles.listItem, styles.loading)}>
      <div className={styles.info}>
        <Loading
          alignment="left"
          height="30px"
          type="skeleton"
          width={`${randomBetween(35, 465)}px`}
        />
        <Loading
          alignment="left"
          height="19px"
          type="skeleton"
          width={`${randomBetween(35, 200)}px`}
        />
      </div>
      <div className={styles.stats}>
        <Loading
          alignment="left"
          className={styles.skeleton}
          height="116px"
          type="skeleton"
          width="173px"
        />
        <Loading
          alignment="left"
          className={styles.skeleton}
          height="116px"
          type="skeleton"
          width="173px"
        />
      </div>
    </Box>
  )
}

export default LoadingListItem
