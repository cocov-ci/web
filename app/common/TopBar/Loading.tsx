import Loading from '../Loading'

import styles from './TopBar.module.scss'

const LoadingToolbar = () => {
  return (
    <div className={styles.topBar}>
      <Loading
        alignment="left"
        count={1}
        height="65px"
        type="skeleton"
        width="300px"
      />
      <Loading
        alignment="right"
        count={1}
        height="50px"
        type="skeleton"
        width="400px"
      />
    </div>
  )
}

export default LoadingToolbar
