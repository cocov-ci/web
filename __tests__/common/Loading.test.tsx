import { render } from '@testing-library/react'

import Loading from 'app/common/Loading'

import '@testing-library/jest-dom'

describe('Loading', () => {
  it('renders Loading spinner', () => {
    const { container } = render(<Loading type="spinner" />)

    expect(container.querySelector('img')).toBeVisible()
  })

  it('renders Loading skeleton', () => {
    const { container } = render(<Loading type="skeleton" />)

    expect(
      container.getElementsByClassName('react-loading-skeleton').length,
    ).toEqual(1)
  })

  it('renders Loading skeleton with count param', () => {
    const { container } = render(<Loading count={3} type="skeleton" />)

    expect(
      container.getElementsByClassName('react-loading-skeleton').length,
    ).toEqual(3)
  })

  it('renders Loading snapshots', () => {
    const { container } = render(<Loading />)
    expect(container).toMatchSnapshot()
  })
})
