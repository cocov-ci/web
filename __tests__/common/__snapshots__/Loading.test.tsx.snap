// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders Loading snapshots 1`] = `
<div>
  <div
    class="loading center"
  >
    <img
      alt="A spinner indicating an ongoing operation"
      class="spinner"
      data-nimg="1"
      decoding="async"
      height="24"
      loading="lazy"
      src="/icons/spinner.svg"
      srcset="/icons/spinner.svg 1x, /icons/spinner.svg 2x"
      style="color: transparent;"
      width="24"
    />
  </div>
</div>
`;
